main.go

```diff
@@ -32,7 +34,13 @@ func checkForErrorsAndExit(err error) {
 		} else {
 			logger.Println(err)
 		}
-		os.Exit(1)
+		// exit with the underlying error code
+		var retCode int
+		if exiterr, ok := errors.Unwrap(err).(*exec.ExitError); ok {
+			status := exiterr.Sys().(syscall.WaitStatus)
```

Comment:
This will work on darwin, linux, and windows (all the terragrunt OS's).

It's not completely portable (plan9 prob won't work) but unfortunately there's [no portable way to get the exit code](https://godoc.org/os#ProcessState.Sys). Fortunately, [windows](https://golang.org/src/syscall/syscall_windows.go#L835) and [*nix ](https://golang.org/src/syscall/syscall_linux.go#L224) have the same method signatures.